sdk/core/azure-core-test/src/main/java/com/azure/core/test/http/PlaybackClient.java

```diff
@@ -141,7 +160,12 @@ private String applyReplacementRule(String text) {
     }
 
     private static String removeHost(String url) {
-        URI uri = URI.create(url);
-        return String.format("%s?%s", uri.getPath(), uri.getQuery());
+        UrlBuilder urlBuilder = UrlBuilder.parse(url);
+
+        if (urlBuilder.query().containsKey("sig")) {
```

Comment:
It's useful to keep the rest of the SAS token the same as that is what is being matched in the playback client, it could help catch scenarios where we accidentally break the generation of it.